src/triage/component/catwalk/model_trainers.py

```diff
@@ -246,7 +258,7 @@ def _write_model_to_db(
         return model_id
 
     def _train_and_store_model(
-        self, matrix_store, class_path, parameters, model_hash, misc_db_parameters, random_seed
+        self, matrix_store, class_path, parameters, model_hash, misc_db_parameters, random_seed, retrain, model_group_id, 
```

Comment:
I'm curious about the addition of 'retrain' and 'model_group_id'. Before adding these, were you trying this and getting different model group ids? You *should* get the same model group id the second time, if the random seed is the same as the first time.

Reply:
since "retrain" is using the new matrix_metadata(we are creating a new matrix) even with the same random_seed, so `self.model_grouper.get_model_group_id()` will return a new `model_group_id`. That's why for "retrain" I have to pass the "model_group_id".

Reply:
Hmmm... I'm not sure I follow here, even if we're creating a new training matrix, shouldn't it have the same metadata (e.g., label, training history, as of date frequency, and so on) as previous training matrices associated with the model group? I think I agree with @thcrock  here that we should definitely expect to get the correct `model_group_id` back from `get_model_group_id()` just as though this was just another training matrix from a train/test split in the original experiment (incidentally, the random seed actually shouldn't come into it either way since that's in the `model_id` rather than the `model_group_id`).

@tweddielin, if you are getting different `model_group_id`s in practice, we should probably dig in a bit here into why the new training matrix metadata is different from the initial run.

Reply:
I agree, ideally we should be getting the same model group ids for comparing. But @tweddielin makes a good point, there are a lot of things in the matrix metadata, and any one thing we change would make it hash differently. It may even be something as simple as the 'matrix type' (train vs test). That would show up in the matrix metadata. IIRC when we first started doing this PR, we chose a new matrix type. If this PR is still using a new matrix type instead of train or test, then I imagine it would get a new model group. 

@tweddielin can you show the output of an example 'previous training matrix' and one from the retraining so we can see what it would take to get them to sync up and get the same model group id?